README.md

```diff
@@ -100,6 +100,280 @@ Feel free to open an issue for questions, problems, or suggestions with our bund
+use App\Dto\ResetPasswordInput;
+use App\Repository\ResetPasswordRequestRepository;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Uid\UuidV4;
+use SymfonyCasts\Bundle\ResetPassword\Model\ResetPasswordRequestInterface;
+use SymfonyCasts\Bundle\ResetPassword\Model\ResetPasswordRequestTrait;
+
+/**
+ * @ApiResource(
+ *     security="is_granted('IS_ANONYMOUS')",
+ *     input=ResetPasswordInput::class,
+ *     output=false,
+ *     shortName="reset-password",
+ *     collectionOperations={
+ *          "post" = {
+ *              "denormalization_context"={"groups"={"reset-password:post"}},
+ *              "status" = 202,
+ *              "validation_groups"={"postValidation"},
+ *          },
+ *     },
+ *     itemOperations={
+ *          "put" = {
+ *              "denormalization_context"={"groups"={"reset-password:put"}},
+ *              "validation_groups"={"putValidation"},
+ *          },
+ *     },
+ * )
+ *
+ * @ORM\Entity(repositoryClass=ResetPasswordRequestRepository::class)
+ */
+class ResetPasswordRequest implements ResetPasswordRequestInterface
+{
+    use ResetPasswordRequestTrait;
+
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="string", unique=true)
+     */
+    private string $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=User::class)
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private User $user;
+
+    public function __construct(User $user, \DateTimeInterface $expiresAt, string $selector, string $hashedToken)
+    {
+        $this->id = new UuidV4();
+        $this->user = $user;
+        $this->initialize($expiresAt, $selector, $hashedToken);
+    }
+
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    public function getUser(): User
+    {
+        return $this->user;
+    }
+}
+```
+
+Because the `ResetPasswordHelper::generateResetToken()` method is responsible for
+creating and persisting a `ResetPasswordRequest` object after the reset token has been
+generated, we can't call `POST /api/reset-passwords` with `['email' => 'someone@example.com']`.
+
+We'll create a Data Transfer Object (`DTO`) first, that will be used by a Data Persister
+to generate the actual `ResetPasswordRequest` object from the email address provided
+in the `POST` api call.
+
+```php
+<?php
+
+namespace App\Dto;
+
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ */
+class ResetPasswordInput
+{
+    /**
+     * @Assert\NotBlank(groups={"postValidation"})
+     * @Assert\Email(groups={"postValidation"})
+     * @Groups({"reset-password:post"})
+     */
+    public string $email;
+
+    /**
+     * @Assert\NotBlank(groups={"putValidation"})
+     * @Groups({"reset-password:put"})
+     */
+    public string $token;
+
+    /**
+     * @Assert\NotBlank(groups={"putValidation"})
+     * @Groups({"reset-password:put"})
+     */
+    public string $plainTextPassword;
```

Comment:
It might be easier to split this into 2 DTO objects - something like `RequestResetPasswordInput` and `ResetPasswordInput`. Then you can configure the `input` on each, specific operation i think (instead of having the `input=` on the top-level.

Another option would be to create these 2 DTO's and make THEM each their own `@ApiResource`... each with 1 operation. I'm not sure if having `ResetPasswordRequest` as the `@ApiResource` is doing us any favors, as we never use it as the input or output. That would be my biggest potential feedback on this otherwise awesome effort. `input` and `output` DTO's are kind of an edge-case feature in API Platform... so if we can find a clean way to do this without them, that might ideal.